### PR TITLE
Nextjs setup example page

### PR DESCRIPTION
**Pull request by danijelbuhin**
Updated nextjs setup example page

### CODE REVIEW

File: docs/examples/nextjs-setup.md [hunk 1]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+
+export interface IStores {
+  store: AppStore;
+}
+
+interface IProviderProps {
+  children: ReactNode;
+}
+
+const storeContext = createContext<IStores>({ store });
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:142]: Where is the store variable coming from?

File: docs/examples/nextjs-setup.md [hunk 2]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+
+const __NEXT_MOBX_STORE__ = 'store';
+
+declare global {
+  interface Window {
+    store: AppStore;
+  }
+}
+
+const isBrowser = !!(
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:68]:
```suggestion
const isBrowser = Boolean(
```

File: docs/examples/nextjs-setup.md [hunk 3]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+}
+```
+<!--JavaScript-->
+
+```jsx
+import { IRawCollection } from 'datx/dist/interfaces/IRawCollection';
+
+const __NEXT_MOBX_STORE__ = 'store';
+
+const isBrowser = !!(
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:97]:
```suggestion
const isBrowser = Boolean(
```

File: docs/examples/nextjs-setup.md [hunk 4]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+      };
+    };
+  }
+
+  return WithDatx;
+};
+```
+<!--JavaScript-->
+
+```tsx
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:246]:
```suggestion
```jsx
```

File: docs/examples/nextjs-setup.md [hunk 5]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+import App from 'next/app';
+import React from 'react';
+
+import { getOrInitializeStore } from '../../stores/getOrInitializeStore';
+import { AppStore } from '../../stores/AppStore';
+import { StoreProvider } from '../stores/utils/useStore';
+
+export const withDatx = (PageComponent, options = { ssr: true }) => {
+  const WithDatx = ({ initialMobxState, ...props }) => {
+    const store: AppStore = getOrInitializeStore(initialMobxState);
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:256]:
```suggestion
    const store = getOrInitializeStore(initialMobxState);
```

File: docs/examples/nextjs-setup.md [hunk 6]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+
+  if (ssr || PageComponent.getInitialProps) {
+    WithDatx.getInitialProps = async (context) => {
+      // Get or Create the store with `undefined` as initialState
+      // This allows you to set a custom default initialState
+      const store = getOrInitializeStore();
+
+      // Provide the store to getInitialProps of pages
+      // Run getInitialProps from HOCed PageComponent
+      const pageProps: any =
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:289]:
```suggestion
      const pageProps =
```

File: docs/examples/nextjs-setup.md [hunk 7]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
...
+export type NextPageWithDatx<P = {}, IP = P> = NextPage<P, IP> & {
+  getInitialProps?: (ctx: PageContextWithStore) => Promise<IP>;
+};
+
+export const withDatx = (PageComponent: NextPageWithDatx, options: IOptions = { ssr: true }) => {
+  const WithDatx = ({ initialMobxState, ...props }: any) => {
+    const store: AppStore = getOrInitializeStore(initialMobxState);
+
+    return (
+      <StoreContext.Provider value={store}>
```
Comment (DarkoKukovec) [inline docs/examples/nextjs-setup.md:198]: Where is `StoreContext` coming from?

File: docs/examples/nextjs-setup.md [hunk 8]
```diff
@@ -3,4 +3,418 @@ id: nextjs-setup
 title: Next.js setup
 ---
 
-> Detailed instructions for Next.js setup will be added soon. In the meantime, you can check out <br> [this gist](https://gist.github.com/isBatak/883b5ae602146848c04f8544a33c368b) for a current code example.
+## Preface
+
+Just like in basic [react-setup](react-setup), we will use the `react-mobx ^6.x` since we will be only using functional components and pages with [hooks](https://reactjs.org/docs/hooks-intro.html).
```
Comment (badurinantun) [inline docs/examples/nextjs-setup.md:8]: Should this be `mobx-react`?

**Comment by danijelbuhin** [comment]
Closing this PR since this is a very old proposal.

A new one will be created with guidelines from the new frontend handbook and new versions of datx and nextjs.

**Pull request closed without merging**